RestApi/Sepes.RestApi/Model/Pod.cs

```diff
@@ -36,10 +45,32 @@ public class Pod
             ImmutableList<User>.Empty, ImmutableList<DataSet>.Empty, ImmutableList<DataSet>.Empty) {}
 
 
-        public string networkName => $"{studyId}-{name.Replace(" ", "-")}-Network";
-        public string subnetName => $"{studyId}-{name.Replace(" ", "-")}-SubNet";
-        public string resourceGroupName => $"{studyId}-{name.Replace(" ", "-")}-ResourceGroup";
-        public string networkSecurityGroupName => $"{studyId}-{name.Replace(" ", "-")}-NetworkSecurityGroup";
-        public string addressSpace => $"10.{1 + id / 256}.{id % 256}.0/24";
+        public override bool Equals(object obj)
+        {
+            return obj is Pod pod &&
+                   id == pod.id &&
+                   name == pod.name &&
+                   studyId == pod.studyId &&
+                   allowAll == pod.allowAll &&
+                   Enumerable.SequenceEqual(incoming, pod.incoming) &&
+                   Enumerable.SequenceEqual(outgoing, pod.outgoing) &&
+                   Enumerable.SequenceEqual(users, pod.users);
+        }
+
+        public override int GetHashCode()
+        {
+            var hash = new HashCode();
+            hash.Add(id);
+            hash.Add(name);
+            hash.Add(studyId);
+            hash.Add(allowAll);
+            hash.Add(incoming);
+            hash.Add(outgoing);
+            hash.Add(users);
```

Comment:
You can't simply take the hash of lists. As that link to standard list equality. Best to not use them as part of the hash.